src/dimension.c

```diff
@@ -441,3 +519,431 @@ hyperspace_calculate_point(Hyperspace *hs, HeapTuple tuple, TupleDesc tupdesc)
 
 	return p;
 }
+
+static inline int64
+interval_to_usec(Interval *interval)
```

Comment:
So, my reasoning is that since this function is only used here, it makes sense to have it here where it has the proper context. If you feel strongly, I can move it to `utils.c`. That said, I think our time-related functions in `utils.c` is a bit of a mess right now. I'd like to refactor that into its own file at some point.